### PR TITLE
♻️ Refactor products plugin before extension

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Some refactoring work before https://github.com/ITISFoundation/osparc-simcore/pull/4739
- runs ruff check --fix
- created `products.api`
- reviewed tests and fixtures
- general cleanup

## Related issue/s

- Preparation for #4546

## How to test

in place

## DevOps

None

### CODE REVIEW

File: services/web/server/src/simcore_service_webserver/products/_db.py [hunk 1]
```diff
@@ -26,8 +26,8 @@ async def iter_products(conn: SAConnection) -> AsyncIterator[ResultProxy]:
     async for row in conn.execute(
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/products/_db.py:26]: does this return empty rows now?

**Review by sanderegg — APPROVED** [review]

File: packages/postgres-database/tests/products/test_utils_products.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 
 import asyncio
-from typing import Callable
+from collections.abc import Callable
```
Comment (matusdrobuliak66) [inline packages/postgres-database/tests/products/test_utils_products.py:9]: Q: just curious, why this change? thanks

**Review by matusdrobuliak66 — APPROVED** [review]

File: packages/postgres-database/tests/products/test_utils_products.py (hunk 2 above)
Comment (pcrespov) [reply packages/postgres-database/tests/products/test_utils_products.py:9]: ruff

File: services/web/server/src/simcore_service_webserver/products/_db.py (hunk 1 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/products/_db.py:26]: You are right. no it does not

**Review by GitHK — APPROVED** [review]
